File: Task 1/C++/Components/Product/Appliance.cpp
#include "Appliance.h"

Appliance::Appliance() : Electronics(){
    size = "";
}


Appliance::Appliance(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, std::string manufacturer, std::string size) 
            : Electronics(name, cost, supplier, discount_amount, packing_units, cost_per_unit, manufacturer){
    this->size = size;
}

Appliance::Appliance(const Appliance& other) : Electronics(other) {
    this->size = other.size;
}

std::string Appliance::display_product_info(){
    return Electronics::display_product_info() + 
            "\nSize: "+size;
}

File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Appliance.java
package com.logistics.Components.Product;

public class Appliance extends Electronics {

    private String size;

    public Appliance() {
        super();
        size = "";
    }

    public Appliance(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, String manufacturer, String size) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit, manufacturer);
        this.size = size;
    }

    // Copy Constructor
    public Appliance(Appliance other) {
        super(other);
        this.size = other.size;
    }

    // Clone method
    @Override
    public Appliance clone() {
        return new Appliance(this);
    }

    // Getter
    public String getSize() { return size; }

    // Setter
    public void setSize(String size) { this.size = size; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nSize: " + size;
    }
}
